packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -19,14 +19,15 @@ const TAG = 'NavigationService'
 type SafeNavigate = typeof navigate
 
 export const navigationRef = createRef<NavigationContainerRef>()
+export const isReadyRef: MutableRefObject<boolean | null> = createRef()
 
 async function ensureNavigator() {
   let retries = 0
-  while (!navigationRef.current && retries < 3) {
+  while (!navigationRef.current && !isReadyRef.current && retries < 5) {
```

Comment:
Can you please make the retries a constant that's defined at the top of the file?